clippy_lints/src/map_clone.rs

```diff
@@ -46,10 +48,15 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                                 // .cloned() only removes one level of indirection, don't lint on more
                                 walk_ptrs_ty_depth(cx.tables.pat_ty(&first_arg.pat)).1 == 1
                             {
-                                span_help_and_lint(cx, MAP_CLONE, expr.span, &format!(
-                                    "you seem to be using .map() to clone the contents of an {}, consider \
-                                    using `.cloned()`", type_name),
-                                    &format!("try\n{}.cloned()", snippet(cx, args[0].span, "..")));
+                                // the argument is not an &mut T
```

Comment:
Nit: Why is this not reusing the `if_let_chain!` above? This could reduce rightwards drift and IMHO make the code a bit more readable.